### PR TITLE
Fix circular references

### PR DESCRIPTION
**Pull request by Itee**
Next to #10627 -> #10630 -> #10633 

Known bugs fixed !
Merge success !
Style apply !
Test done !

PR Ready...

### CODE REVIEW

**Comment by mrdoob** [comment]
Seems like there are no regressions this time! Thanks!

File: src/animation/AnimationAction.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ function AnimationAction( mixer, clip, localRoot ) {
 
 }
 
-AnimationAction.prototype = {
+Object.assign( AnimationAction.prototype, {
 
 	constructor: AnimationAction,
```
Comment (mrdoob) [inline src/animation/AnimationAction.js]: We probably don't need the `contructor:` lines anymore, do we?

File: src/core/BufferAttribute.js [hunk 2]
```diff
@@ -32,7 +32,17 @@ function BufferAttribute( array, itemSize, normalized ) {
 
 }
 
-BufferAttribute.prototype = {
+Object.defineProperty( BufferAttribute.prototype, "needsUpdate", {
+
+	set: function(value) {
```
Comment (mrdoob) [inline src/core/BufferAttribute.js]: Hmm, I guess you forgot to remove the `needsUpdate` setter from the `Object.assign()` down bellow. It seems to be compiling fine though?

File: src/animation/AnimationAction.js (hunk 1 above)
Comment (TristanVALCKE) [inline src/animation/AnimationAction.js]: Yes we could remove the constructor line, it is not necessary now. I will give it a try to be sure there is no hidden regression, and propose a new PR.

File: src/core/BufferAttribute.js (hunk 2 above)
Comment (TristanVALCKE) [inline src/core/BufferAttribute.js]: Yes i forgot it ! I will remove it...

**Pull request merged**